### PR TITLE
My profile markup

### PR DESCRIPTION
**Pull request by pavelkortp**

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -33,12 +35,17 @@ android {
...
     }
+
+    buildFeatures{
+        viewBinding = true
+    }
 }
 
 dependencies {
-
+    implementation("com.github.bumptech.glide:glide:4.16.0")
```
Comment (hryhorii11) [inline app/build.gradle.kts]: versions should be expoted to variable

File: app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.transition.Slide
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import com.pavelkortp.socialnetwork.databinding.ActivityAuthBinding
+
+class AuthActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityAuthBinding
```
Comment (hryhorii11) [inline app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt]: I recommend using lazy initialization for binding

File: app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        }
+        toMainActivity(email, pass)
+        return false
+    }
+
+
+    private fun saveData(email: String, pass: String) {
+        val sharedPreferences = getSharedPreferences("my_app_preferences", Context.MODE_PRIVATE)
+        val editor = sharedPreferences.edit()
+        editor.putString("email",email)
```
Comment (hryhorii11) [inline app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt]: export preferences key to constant

File: app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        return false
+    }
+
+
+    /**
+     * Checks if password is invalid
+     */
+    private fun isInvalidPassword(password: String): Boolean {
+        return if (password.isBlank()) {
+            binding.PasswordInput.error = "Password is empty"
```
Comment (hryhorii11) [inline app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt]: export to string resource

File: app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        } else
+            false
+    }
+
+    /**
+     * Checks if email is invalid
+     */
+    private fun isInvalidEmail(email: String): Boolean {
+        return if (email.isBlank()) {
+            binding.EmailInput.error = "Email is empty"
```
Comment (hryhorii11) [inline app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt]: also export to string resource

File: app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            toMainActivity(email!!, pass!!)
+            finish()
+        }
+    }
+
+    private fun toMainActivity(email: String, password: String){
+        val i = Intent(this, MainActivity::class.java)
+        with(i) {
+            putExtra("email", email)
+            putExtra("password", password)
```
Comment (hryhorii11) [inline app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt]: why are you passing the password in the MainActivity?

File: app/src/main/java/com/pavelkortp/socialnetwork/LoginActivity.kt
Comment (hryhorii11) [inline app/src/main/java/com/pavelkortp/socialnetwork/LoginActivity.kt:1]: it seems to me that you should not implement loginActivity for first level

File: app/src/main/java/com/pavelkortp/socialnetwork/LoginActivity.kt [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+class LoginActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityLoginBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.LoginBTN.setOnClickListener {
```
Comment (hryhorii11) [inline app/src/main/java/com/pavelkortp/socialnetwork/LoginActivity.kt:19]: i recommend create method for setting listeners to decompose onCreate method

File: app/src/main/java/com/pavelkortp/socialnetwork/LoginActivity.kt [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+        window.enterTransition = Slide()
+        window.exitTransition = Slide()
+    }
+
+    private fun login(): Boolean {
+        val email = binding.EmailInput.text.toString()
+        val pass = binding.PasswordInput.text.toString()
+
+        if (email.isBlank()) {
```
Comment (hryhorii11) [inline app/src/main/java/com/pavelkortp/socialnetwork/LoginActivity.kt]: this code is repeated in the activity, maybe it makes sense to create a class for validation

File: app/src/main/java/com/pavelkortp/socialnetwork/MainActivity.kt [hunk 9]
```diff
@@ -1,11 +1,56 @@
...
+            "my_app_preferences",
+            Context.MODE_PRIVATE
+        )
+            .edit()
+            .clear()
+            .apply()
+    }
+
+    @Parcelize
+    class State(var counter: Int) : Parcelable
```
Comment (hryhorii11) [inline app/src/main/java/com/pavelkortp/socialnetwork/MainActivity.kt]: export class to another file

File: app/src/main/res/drawable-mdpi/bob.gif
Comment (hryhorii11) [inline app/src/main/res/drawable-mdpi/bob.gif:1]: )

File: app/src/main/res/layout/activity_auth.xml [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/default_margin"
+        android:text="@string/email"
+        android:textColor="@color/Gray"
+        android:textSize="14sp"
+        app:layout_constraintBottom_toTopOf="@id/EmailInput"
+        app:layout_constraintStart_toStartOf="@id/EmailInput" />
+
+    <EditText
```
Comment (hryhorii11) [inline app/src/main/res/layout/activity_auth.xml]: use textInputLayout instead of editText

File: app/src/main/res/layout/activity_auth.xml [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+    android:background="@color/MainBackground"
+    tools:context=".AuthActivity">
+
+    <TextView
+        android:id="@+id/AuthTitle"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/auth_title"
+        android:textColor="@color/White"
+        android:textSize="24sp"
```
Comment (hryhorii11) [inline app/src/main/res/layout/activity_auth.xml]: all sp or dp values should be be exported to dimens

File: app/src/main/res/layout/activity_login.xml [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/AuthTitle" />
+
+
+    <TextView
+        android:id="@+id/EmailLabel"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Email"
```
Comment (hryhorii11) [inline app/src/main/res/layout/activity_login.xml]: export all text to string resources

**Review by hryhorii11 — CHANGES_REQUESTED** [review]

File: app/build.gradle.kts (hunk 1 above)
Comment (pavelkortp) [reply app/build.gradle.kts]: Only glide? Or should others also be included in separate variables? Where should these variables then be stored?

File: app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt (hunk 2 above)
Comment (pavelkortp) [reply app/src/main/java/com/pavelkortp/socialnetwork/AuthActivity.kt]: What`s the benefit of this approach?

**Pull request merged**